sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -101,7 +101,8 @@ static RT::PiProgram createBinaryProgram(const ContextImplPtr Context,
   Plugin.call<PiApiKind::piPlatformGetInfo>(Platform, PI_PLATFORM_INFO_NAME,
                              PlatformName.size(), PlatformName.data(), nullptr);
   if (PlatformNameSize > 0u &&
-      std::strncmp(PlatformName.data(), "NVIDIA CUDA", PlatformNameSize) == 0) {
+      std::strncmp(PlatformName.data(), "NVIDIA CUDA BACKEND",
```

Comment:
There are 3 occurrence of this literal string, can you turn it into a macro at least?

Reply:
Shouldn't the lines 91-108 become a run-time check that the active selected BE (Plugin.getBackend()) is CUDA?

Reply:
> There are 3 occurrence of this literal string, can you turn it into a macro at least?

Yes, but I am unsure where to set that macro that is shared between the SYCL RT and the PI CUDA plugin, but doesnt create a dependency on the SYCL RT to a header in the CUDA plugin. 

Maybe I can just add it to the common pi header as a CUDA extension.

Reply:
> Shouldn't the lines 91-108 become a run-time check that the active selected BE (Plugin.getBackend()) is CUDA?

Yes, but that should be on a separate patch I think

Reply:
> Yes, but that should be on a separate patch I think

Why? If you do so then you don't need many of the changes you are doing in this PR.

Reply:
I've changed it to use the `is_cuda` check from platform. Plugin.getBackend() is not yet on sycl branch.

Reply:
getBackend() is in now, please use it instead